api/v1beta1/openstackdataplanenode_types.go

```diff
@@ -72,6 +72,10 @@ type NodeSection struct {
 	// +kubebuilder:validation:Optional
 	// AnsibleVars for configuring ansible
 	AnsibleVars string `json:"ansibleVars,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// AnsiblePrivateSSHKey Private SSH Key for connecting to node
+	AnsibleSSHPrivateKey string `json:"ansibleSSHPrivateKey,omitempty"`
```

Comment:
+1

I think we could just ask the secret name and get the key from there:
`AnsibleSSHSecretName`

Reply:
sorry if this does not fit the current deploy workflow, but could just generate a key pair and store it in a secret, we used [1]  in ospdo do do it. when metal3 is used to deploy the nodes it can be added via cloud-init.

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/pkg/common/ssh.go

Reply:
we can't generate a keypair for the pre-provisioned node case.  I could change it to just take a secret name, and require the user to create the secret ahead of time, but I don't understand the difference between doing that which requires passsing the key in plain text to a Secret vs. passing it in plain text in our custom CRD.

Either way, anyone with access to the namespace, can get the secret contents.